### PR TITLE
Issue #5296: Add php lint to our GitHub Actions workflow [skip tests]

### LINKED ISSUE #5296: Consider php lint for Github Actions
**Description of the need**

It doesn't happen often, but sometimes a syntax error sneaks in.
It can be a more obvious one that only got overlooked in a very big PR, for instance in an automated test. It can be a more subtle one, that only strikes on older PHP versions <s>(short array syntax, using empty() with non-variables...)</s> (the oldest officially supported PHP version is now 5.6.)

Anyway, it would be so easy to prevent that by simply running `php -l` on all changed files, for the oldest and newest PHP Backdrop supports.

**Additional information**

Not sure if this works as-is in the runners, but here's a simple bash one-liner that works for me locally:

```
git diff --name-only 1.x... | while read LINE; do php -l $LINE; done;
```

Feedback is welcome. :smiley:

### PR DESCRIPTION
**Pull request by indigoxela**
Fixes https://github.com/backdrop/backdrop-issues/issues/5296

### CODE REVIEW

**Comment by indigoxela** [comment]
Wrong approach, have to think it over...

File: .github/misc/phplinter.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * Runs the php linter against changed files.
+ *
+ */
+
+/**
+ * Get a list of changed files via git diff and run the linter function.
+ */
+function lint_changed_files() {
+  $exit_code = 0;
+  $descriptorspec = array(
```
Comment (klonos) [inline .github/misc/phplinter.php]:
```suggestion
  $descriptor_spec = array(
```

File: .github/misc/phplinter.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    'php',
+    'inc',
+    'module',
+    'install',
+    'test',
+    'profile',
+  );
+  // Note that "origin" is just a naming convention.
+  $git_command = 'git diff --name-only --diff-filter=d origin/1.x';
+  $git_process = proc_open($git_command, $descriptorspec, $pipes);
```
Comment (klonos) [inline .github/misc/phplinter.php]:
```suggestion
  $git_process = proc_open($git_command, $descriptor_spec, $pipes);
```

File: .github/misc/phplinter.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Run the php lint command and evaluate output.
+ *
+ * Translates the output to a format, GitHub Actions can use for annotations.
+ *
+ * @param string $file
+ *   Path to php file that changed compared to 1.x branch.
+ */
+function phplinter($file) {
+  $error = NULL;
+  $descriptorspec = array(
```
Comment (klonos) [inline .github/misc/phplinter.php]:
```suggestion
  $descriptor_spec = array(
```

File: .github/misc/phplinter.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    1 => array('pipe', 'w'),
+    2 => array('pipe', 'w'),
+  );
+  $lint_command = "php -l $file";
+  $lint_process = proc_open($lint_command, $descriptorspec, $lint_pipes);
```
Comment (klonos) [inline .github/misc/phplinter.php]:
```suggestion
  $lint_process = proc_open($lint_command, $descriptor_spec, $lint_pipes);
```

Comment (indigoxela) [reply .github/misc/phplinter.php]: Many thanks @klonos, suggestions committed. :+1:

**Pull request closed without merging**